Pyomo 5.6 CPLEX "Unallowed character (:) found in CPLEX log file" ValueError on Windows

Hi,

After installing Pyomo 5.6, my unit tests fail on Windows 7. Here are the details:

filename = 'C:\\Users\\<username>\\AppData\\Local\\Temp\\tmp9v95r1qz.cplex.log'
description = 'log'

    def _validate_file_name(cplex, filename, description):
        """Validate filenames against the set of allowable chaacters in CPLEX.

        Returns the filename, possibly enclosed in double-quotes, or raises
        a ValueError is unallowable characters are found.
        """
       ...
E           ValueError: Unallowed character (:) found in CPLEX log file path/name.
E               For portability reasons, only [a-zA-Z0-9 .-_] are allowed.

..\..\AppData\Local\Continuum\Anaconda3\lib\site-packages\pyomo\solvers\plugins\solvers\CPLEX.py:53: ValueError

It seems that "C:\\path..." is no longer valid in Pyomo 5.6, which is a bit difficult on Windows.

Others are experiencing the same problem:

https://stackoverflow.com/questions/54095681/why-am-i-getting-an-unallowed-character-error-in-pyomo-with-cplex

Can you fix this?

Thanks.

--- a/pyomo/solvers/plugins/solvers/CPLEX.py
+++ b/pyomo/solvers/plugins/solvers/CPLEX.py
@@ -62,7 +62,7 @@ def _validate_file_name(cplex, filename, description):
                 "CPLEX or remove the space from the %s path."
                 % (description, filename, description))
     return filename
-_validate_file_name.allowed_characters = r"a-zA-Z0-9 \.\-_\%s" % (os.path.sep,)
+_validate_file_name.allowed_characters = r"a-zA-Z0-9 :\.\-_\%s" % (os.path.sep,)
 _validate_file_name.illegal_characters = re.compile(
     '[^%s]' % (_validate_file_name.allowed_characters,))
 

